akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1588,21 +1591,19 @@ final class TakeWithin[T](val timeout: FiniteDuration) extends SimpleLinearGraph
 final class DropWithin[T](val timeout: FiniteDuration) extends SimpleLinearGraphStage[T] {
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) with InHandler with OutHandler {
 
-    private var allow = false
+    override def preStart(): Unit = scheduleOnce("DropWithinTimer", timeout)
```

Comment:
Does that actually need the timer at all? Wouldn't it be simpler (and cheaper) to check the end of drop period in `onPush`?

(This also points to the other issue that for this (and maybe other) timing-related stages there could be alternative implementation that would eagerly pull and drop elements during the initial drop phase.)